src/bin/ion/commands/beta/generate/context.rs

```diff
@@ -0,0 +1,60 @@
+use crate::commands::beta::generate::utils::Language;
+use serde::Serialize;
+use tera::Tera;
+
+/// Represents a context that will be used for code generation
+pub struct CodeGenContext {
+    // Represents the templating engine - tera
+    pub(crate) tera: Tera,
+    pub(crate) language: Language,
+    // Initially the data_model field is set to None.
+    // Once an ISL type definition is mapped to a data model this will have Some value.
+    pub(crate) data_model: Option<DataModel>,
```

Comment:
We only get a fully baked `CodeGenContext` when we determine a data model. And that is done after we process a type definition or atleast one constraint from that type definition. I am not sure if I follow what you mean by separate builder? Did you mean something use another context object for initialization?

Reply:
Having read a bit further now, I'm actually a bit less sure of the intent of this field. The type is `DataModel`, which I thought would be a collection of types that were used throughout codegen, but I think it's actually the abstract type (`Scalar`, `Sequence`, or `Struct`) that a single ISL type definition is mapped to. The field gets set and unset several times during the course of codegen. It may be easier for me to understand if we talk this through offline.

Reply:
With new change, context only stores data model property and is locally created for
each ISL type code generation.